### DISCUSSION TITLE
FavIcon vom Profi für den OpenDTU-Alias

### DISCUSSION THREAD

**Discussion by designermohr** [tbnobody/OpenDTU#1605] [Ideas]
Hallo Entwickler-Community des OpenDTU. Ich bin auch ein Besitzer einer PV und nutze euer OpenDTU-System. Wie ich aber gesehen habe ist das FavIcon auf den Apple/iOS Systemen "pixelig".
(FavIcon = Die Grafik die auf dem mobilen Endgeräten angelegt wird, wenn man einen Direktlink/Alias auf die Webadresse legt.)
Mein Angebot: Ein FavIcon vom Profi. Ich bin Grafiken und kann euch SVG-FavIcons anlegen. (Vektor-Grafiken) Diese sind dann skalierbar und nicht mehr pixelig.

**Comment by stefan123t** [comment]
@designermohr danke für das Angebot, das FavIcon ist [hier](https://github.com/tbnobody/OpenDTU/tree/master/webapp/public) aktuell nur als favicon.ico und favicon.png vorhanden.

Ich habe es mal anhand eines "freien" 512x512 Images in ein SVG Pfad konvertiert (siehe unten)

Die Vorlage scheint unter einer Creative Commons Attribution-NoDerivs 3.0 Lizenz von icons8 zu stehen.
Auf Anfrage stellen sie die Icons aber wohl auch für OpenSource Projekte unter einer passenden Lizenz zur Verfügung. 

Das Problem neben der Lizensierung ist eventuell auch noch die richtigen Einbindung in den Meta Tags der [index.html](https://github.com/tbnobody/OpenDTU/blob/master/webapp/index.html) und dem site.webmanifest:
```
    <link rel="icon" href="/favicon.ico" />
    <link rel="shortcut icon" type="image/png" href="/favicon.png">
    <link rel="apple-touch-icon" href="/favicon.png">
    <link rel="manifest" href='/site.webmanifest' />
```

Würdest Du ein extra Icon(set) für OpenDTU designen und unter der GPL zur Verfügung stellen ?

![solar-panel](https://github.com/tbnobody/OpenDTU/assets/11159753/57262035-bb59-4e38-a812-cb17d98a5595)

**Reply by designermohr** [reply]
Ja gerne. Ich würde gerne etwas am Projekt OpenDTU und für die Community beitragen. Auch ein neues Design für ein Favicon.

**Reply by designermohr** [reply]
Im oben gezeigten HTML-Beispiel werden nur ein Teil der System mit optimalen FavIcon "bedient". Leider ist durch die Vielzahl der mobilen Endgeräte mit sehr unterschiedlicher Auflösung auch eine Vielzahl (mindestens 4 manchmal mehr) von Varianten des FavIcon in unterschiedlichen Größen notwendig. (152x152 Pixel, 180x180 ....) ... 
Dies kann ich gerne auf Wunsch erstellen.
Frage: Ist eine richtige Kachel, also ein Quadrat mit abgerundeten Ecken, für das Touch-Icon auf den mobilen Endgeräten gewünscht? Sieht natürlich chic aus. Das ist für mich kein wesentlicher Mehraufwand!

**Reply by stefan123t** [reply]
Das Problem mit verschiedenen Auslösungen ist, dass wir das irgendwie speichereffizient in den Header der index.html bekommen müssen und auch nicht wirklich verschiedene Auslösungen des favicons vorhalten können, da selbst das Flash Memory des ESP32 begrenzt ist. Daher wäre ein SVG das alle Endgeräte passend skalieren natürlich von Vorteil gegenüber einem statisch gerasterten ICO / PNG file.
Man m.E. kann evtl. noch etwas Memory sparen, wenn man das Icon als gzip-compressed speichert und mit den richtigen Headern (DEFLATE) ausliefert.
Ob das mit runden Ecken oder ohne besser aussieht weißt Du sicher selbst am Besten, wie gesagt wir können nicht allzu viele Varianten vorsehen, vielleicht ist es auch einfach freigestellt mit transparentem Hintergrund und weißem Rand wie o.a. schon schick.
Aber entscheiden muß es sowieso @tbnobody wenn der Aufwand (Speicher) über dem Nutzen (je Endgerät eigene Skalierung) liegt, kann es auch sein dass er es gar nicht einbaut.

**Reply by tbnobody** [reply]
Aktuell belegen beide, .ico und .png < 8kb. Ich würde ungern für etwas mehr Auflösung mehr Speicher vergeben. Lieber in mehr reele Features stecken. Ich muss sagen, mir war bisher nicht bewusst, das ein favicon.ico welches mit dem tag `<link rel="icon" href="/favicon.ico" />` auch ein SVG sein kann. War bisher der Annahme dass das immer ein ico als mime-type braucht.

**Reply by designermohr** [reply]
Oh. Verstehe. Aus der Perspektive habe ich es gar nicht betrachtet. 
Ok. Dann arbeite ich eine SVG-Version mit Kachel und runden Ecken auf der Basis des vorhandenen SVG-Icon mal aus. 
Melde mich sobald ich ein Ergebnis habe.

**Reply by stefan123t** [reply]
@designermohr kannst ja das o.a. SVG als Ausgangsbasis verwenden. Das sollte relativ einfach sein und somit wenig Speicherplatz verbrauchen. Evtl. noch wie von Dir vorgeschlagen runde Ecken und einen Farbverlauf in den Hintergrund oder so, dann sieht es bestimmt schick aus.

@tbnobody ob das favicon.ico auch ein SVG sein kann das weiß ich auch nicht, ich war auch Deiner Annahme, daß da nur ein klassisches Windows ICO Format geht. Aber das ist Hörensagen der letzten 20-30 Jahre, vielleicht hat sich das ja auch in der Zwischenzeit geändert.
Hast Du dazu etwas gefunden, welche Browser bzw. iOS / Android das unterstützen ?

**Comment by designermohr** [comment]
Hallo Entwickler-Community des OpenDTU.
Sorry für die wirklich späte Rückmeldung.

Ich habe aber keine optimale Lösung erzeugen kann, aber nach meiner Meinung immerhin nutzbare.
Betrachtet hierzu die folgenden Links zur Vorschau und Download.
Einmal sind die neuen FavIcons im Content im Browser vergrößert dargestellt aber auch direkt in den HTML-Header der jeweiligen Seiten eingearbeitet, so dass mann die FavIcon in Anwendung im Reiter, Lesezeichen etc. beurteilen kann.

Gebt mir bitte auch eine Rückmeldung für Verbesserungsvorschläge etc.

# das überarbeitete OpenDTU FavIcon
im Original-Design jedoch nun als SVG und optional als PNG
Vorschau: 
http://opendtu.motyp.de/

Download:
http://opendtu.motyp.de/apple-touch-icon-precomposed.png
http://opendtu.motyp.de/favicon-180x180-full-white-color.svg
http://opendtu.motyp.de/favicon.png


# zwei weitere Design-Vorschläge
Ich habe mir erlaubt, auch eine Optimierung am Design zu tätig.

## New Design 1
Vorschau:
http://opendtu.motyp.de/new-design-1/index.html

Download:
http://opendtu.motyp.de/new-design-1/favicon-180x180-full-white-color-new-sun-3-4.svg


## New Design 2
Vorschau:
http://opendtu.motyp.de/new-design-2/index.html

Download:
http://opendtu.motyp.de/new-design-2/favicon-180x180-full-white-color-thinner-lines-3-7.svg
http://opendtu.motyp.de/new-design-2/apple-touch-icon-precomposed.png
http://opendtu.motyp.de/new-design-2/favicon.png

**Comment by designermohr** [comment]
![iPhone Screen](https://github.com/tbnobody/OpenDTU/assets/94675539/7bf53ca7-20d0-43b0-a993-d50a227d0f53)
